src/mtype.d

```diff
@@ -7803,14 +7803,11 @@ public:
         {
             printf("TypeStruct::dotExp(e = '%s', ident = '%s')\n", e.toChars(), ident.toChars());
         }
+
         // Bugzilla 14010
         if (ident == Id._mangleof)
             return getProperty(e.loc, ident, flag);
-        if (!sym.members)
-        {
-            error(e.loc, "struct %s is forward referenced", sym.toChars());
-            return new ErrorExp();
-        }
```

Comment:
Should add an assert, then. Asserts are cheap, they document assumptions, and check them. dmd is very complex already, and so this is doubly important.